smartsim/_core/launcher/step/slurm_step.py

```diff
@@ -200,18 +207,21 @@ def _build_exe(self) -> t.List[str]:
         if self._get_mpmd():
             return self._make_mpmd()
 
-        exe = self.run_settings.exe
-        args = self._get_exe_args_list(self.run_settings)
+        exe = self.entity.exe
+        args = self._get_exe_args_list(self.entity)
         return exe + args
 
+    # There is an issue here, exe and exe_args are no longer attached to the runsettings
+    # This functions is looping through the list of run_settings.mpmd and build the variable
+    # cmd
     def _make_mpmd(self) -> t.List[str]:
         """Build Slurm multi-prog (MPMD) executable"""
-        exe = self.run_settings.exe
-        args = self._get_exe_args_list(self.run_settings)
+        exe = self.entity.exe
+        args = self._get_exe_args_list(self.entity)
         cmd = exe + args
 
         compound_env_vars = []
-        for mpmd_rs in self._get_mpmd():
+        for mpmd_rs in self._get_mpmd():  # returns a list of runsettings
```

Comment:
Just trying to remove some potential "tech-debt-y" type comments, while we are here!
```suggestion
        for mpmd_rs in self._get_mpmd():
```